### PR TITLE
Document the process for maintaining (quarkus | camel)-master branches

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: docs/modules/ROOT/pages/contributor-guide/ci.adoc [hunk 1]
```diff
@@ -11,6 +11,28 @@
 |quarkus-master| To verify camel-quarkus against the latest Quarkus SNAPSHOT release
 |===
 
+=== Branch maintenance
+
+Some scheduled build jobs (mentioned below) run each day to synchronize branches `camel-master` & `quarkus-master` with the latest work from the `master` branch.
+
+Sometimes these jobs fail and it is necessary to fix any issues and manually rebase the branches. The process for doing this is as follows.
+
+1. Depending on which branch you're rebasing, you may want to begin by building the latest Camel or Quarkus master branch to avoid pulling in non-deterministic SNAPSHOT artifacts. Otherwise you can build the branches and activate the relevant SNAPSHOT repositories via Maven profiles `-Papache-snapshots` or `-Poss-snapshots`.
```
Comment (zbendhiba) [inline docs/modules/ROOT/pages/contributor-guide/ci.adoc:20]: I believe in the Quarkus project, the branch is named `main` now

Comment (jamesnetherton) [reply docs/modules/ROOT/pages/contributor-guide/ci.adoc]: Yes - good spot. I updated the wording for that in the latest commit.

File: docs/modules/ROOT/pages/contributor-guide/ci.adoc [hunk 2]
```diff
@@ -11,6 +11,28 @@
 |quarkus-master| To verify camel-quarkus against the latest Quarkus SNAPSHOT release
 |===
 
+=== Branch maintenance
+
+Some scheduled build jobs (mentioned below) run each day to synchronize branches `camel-master` & `quarkus-master` with the latest work from the `master` branch.
+
+Sometimes these jobs fail and it is necessary to fix any issues and manually rebase the branches. The process for doing this is as follows.
+
+1. Depending on which branch you're rebasing, you may want to begin by building the latest Camel `master` or Quarkus `main` branch to avoid pulling in non-deterministic SNAPSHOT artifacts. Otherwise you can build the branches and activate the relevant SNAPSHOT repositories via Maven profiles `-Papache-snapshots` or `-Poss-snapshots`.
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/contributor-guide/ci.adoc:20]: Pro tip: Build Quarkus with `mvnd clean install -Dquickly` and Camel with `mvnd clean install -Pfastinstall`

Comment (lburgazzoli) [reply docs/modules/ROOT/pages/contributor-guide/ci.adoc:20]: maybe one day we should use the same profile anme across the projects :)

**Review by ppalaga — APPROVED** [review]

**Pull request merged**